conduit/plugins/importers/algod/algod_importer.go

```diff
@@ -414,69 +417,122 @@ func (algodImp *algodImporter) getDelta(rnd uint64) (sdk.LedgerStateDelta, error
 	return delta, nil
 }
 
-func (algodImp *algodImporter) GetBlock(rnd uint64) (data.BlockData, error) {
-	var blockbytes []byte
-	var err error
-	var status models.NodeStatus
-	var blk data.BlockData
+// SyncError is used to indicate algod and conduit are not synchronized.
+type SyncError struct {
+	rnd      uint64
+	expected uint64
+}
 
-	for r := 0; r < retries; r++ {
-		status, err = algodImp.aclient.StatusAfterBlock(rnd - 1).Do(algodImp.ctx)
-		algodImp.logger.Tracef("importer algod.GetBlock() called StatusAfterBlock(%d) err: %v", rnd-1, err)
-		if err != nil {
-			// If context has expired.
-			if algodImp.ctx.Err() != nil {
-				return blk, fmt.Errorf("GetBlock ctx error: %w", err)
-			}
-			err = fmt.Errorf("error getting status for round: %w", err)
-			algodImp.logger.Errorf("error getting status for round %d (attempt %d): %s", rnd, r, err.Error())
-			continue
+func (e *SyncError) Error() string {
+	return fmt.Sprintf("wrong round returned from status for round: %d != %d", e.rnd, e.expected)
+}
+
+func waitForRoundWithTimeout(ctx context.Context, l *logrus.Logger, c *algod.Client, rnd uint64, to time.Duration) (uint64, error) {
+	ctxWithTimeout, cf := context.WithTimeout(ctx, to)
+	defer cf()
+	status, err := c.StatusAfterBlock(rnd - 1).Do(ctxWithTimeout)
+	l.Tracef("importer algod.waitForRoundWithTimeout() called StatusAfterBlock(%d) err: %v", rnd-1, err)
+
+	if err == nil {
+		// On a timeout StatusAfterBlock returns the current status, so check that we've actually reached the correct round.
```

Comment:
I'm attempting to clarify that I'm referring to the server-side timeout, and that this check should be technically unnecessary because our local timeout is less than the server-side timeout.